Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/events/server/qrcodepostertemplate/QrCodePosterTemplateServer.kt

```diff
@@ -1,60 +1,66 @@
-            // TODO return cached or default response
-            throw HttpException(response)
-        }
-        if (response.body() == null) {
-            throw IllegalStateException("Response is successful, but body is empty.")
+        return try {
+            QrCodePosterTemplate.QRCodePosterTemplateAndroid.parseFrom(binaryTemplate)
+        } catch (exception: InvalidProtocolBufferException) {
+            throw QrCodePosterTemplateInvalidResponseException(
+                message = "QR Code poster template could not be parsed",
+                cause = exception
+            )
         }
+    }
 
-        val fileMap = response.body()!!.byteStream().unzip().readIntoMap()
-
-        val exportBinary = fileMap[EXPORT_BINARY_FILE_NAME]
-        val exportSignature = fileMap[EXPORT_SIGNATURE_FILE_NAME]
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    suspend fun getTemplateFromApiOrCache(): ByteArray {
+        return try {
+            val response = api.getQrCodePosterTemplate()
+            if (response.body() == null) {
+                throw IllegalStateException("Response is successful, but body is empty.")
+            }
 
-        if (exportBinary == null || exportSignature == null) {
-            throw QrCodePosterTemplateInvalidResponseException(message = "Unknown files: ${fileMap.keys}")
-        }
+            val fileMap = response.body()!!.byteStream().unzip().readIntoMap()
 
-        val hasValidSignature = signatureValidation.hasValidSignature(
-            exportBinary,
-            SignatureValidation.parseTEKStyleSignature(exportSignature)
-        )
+            val exportBinary = fileMap[EXPORT_BINARY_FILE_NAME]
+            val exportSignature = fileMap[EXPORT_SIGNATURE_FILE_NAME]
 
-        if (!hasValidSignature) {
-            throw QrCodePosterTemplateInvalidResponseException(message = "Invalid Signature!")
-        }
+            if (exportBinary == null || exportSignature == null) {
+                throw QrCodePosterTemplateInvalidResponseException(message = "Unknown files: ${fileMap.keys}")
+            }
 
-        return try {
-            QrCodePosterTemplate.QRCodePosterTemplateAndroid.parseFrom(exportBinary)
-        } catch (exception: InvalidProtocolBufferException) {
-            throw QrCodePosterTemplateInvalidResponseException(
-                message = "QR Code poster template could not be parsed",
-                cause = exception
+            val hasValidSignature = signatureValidation.hasValidSignature(
+                exportBinary,
+                SignatureValidation.parseTEKStyleSignature(exportSignature)
             )
+
+            if (!hasValidSignature) {
+                throw QrCodePosterTemplateInvalidResponseException(message = "Invalid Signature!")
+            }
+
+            exportBinary
+        } catch (exception: Exception) {
+            Timber.e(exception, "Response is not successful, trying to load template from cache")
```

Comment:
Since you already handled the exception, maybe `Timber.d(e,"Message")` so in the logs it is distinguished for uncaught errors

Reply:
ok